File: backend/app/types/auth.types.go
package types

import (
	"time"

	"github.com/google/uuid"
)

// LoginDTO defined the /login payload
type LoginDTO struct {
	Email    string `json:"email" validate:"required,email"`
	Password string `json:"password" validate:"password"`
	Usertype string `json:"userType" validate:"required"`
}

// CreateUserDTO defined the /login payload
type CreateUserDTO struct {
	LoginDTO
	Name     string    `json:"name" validate:"required,min=3"`
	CNPJ     string    `json:"cnpj" validate:"required"`
	Industry []string  `json:"industry" validate:"required"`
	Status   StatusDTO `json:"status"`
}

type StatusDTO struct {
	status string
}
type UpdateUserDTO struct {
	LoginDTO
	ID          string   `json:"id" validate:"required"`
	Name        string   `json:"name"`
	CNPJ        string   `json:"cnpj" `
	Usertype    string   `json:"userType"`
	Industry    []string `json:"industry"`
	NewPassword string   `json:"newpassword"`
}

type UserResponse struct {
	ID         uuid.UUID            `json:"id"`
	Name       string               `json:"name"`
	CNPJ       string               `json:"cnpj"`
	Email      string               `json:"email"`
	Industry   []string             `json:"industry"`
	Usertype   string               `json:"userType"`
	UserStatus []UserStatusResponse `json:"userStatuses"`
}

type UserStatusResponse struct {
	ID        uuid.UUID `json:"statusId"`
	UserID    uuid.UUID `json:"-"`
	Status    string    `json:"userStatus"`
	CreatedAt time.Time `json:"createdAt"`
}

// AccessResponse todo
type AccessResponse struct {
	Token string `json:"token"`
}

// AuthResponse todo
type AuthResponse struct {
	User *UserResponse   `json:"user"`
	Auth *AccessResponse `json:"auth"`
}

// TokenPayload defines the payload for the token
type TokenPayload struct {
	ID       uuid.UUID `json:"id"`
	Email    string    `json:"email"`
	CNPJ     string    `json:"cnpj"`
	Name     string    `json:"name"`
	Industry []string  `json:"industry"`
	Usertype string    `json:"userType"`
}

File: frontend/CorrenteDoBem/src/types/AuthTypes.ts
// src/types/AuthTypes.ts

// Tipos relacionados ao login
export interface LoginDTO {
    email: string;
    password: string;
    userType: string;
  }
  
  // Tipos para os dados de criação de usuário (inclui LoginDTO)
  export interface CreateUserDTO extends LoginDTO {
    name: string;
    cnpj: string;
    industry: string[];
  }

  // Tipos para status do usuário
  export interface UserStatusResponse {
    statusId: string;   // UUID como string
    userId: string;     // UUID como string
    userStatus: string;
    createdAt: string;  // Data no formato ISO (string)
  }
  
  // Tipos para os dados do usuário
  export interface UserResponse {
    id: string;         // UUID como string
    name: string;
    cnpj: string;
    email: string;
    industry: string[];
    userType: string;
    userStatuses: UserStatusResponse[];
  }
  
  
  // Tipos para o token de autenticação
  export interface AccessResponse {
    token: string;  // Token JWT
  }
  
  // Resposta de autenticação
  export interface AuthResponse {
    user: UserResponse;  // Dados do usuário
    auth: AccessResponse; // Token JWT
  }
  
  // Tipos para os dados armazenados no JWT (Payload)
  export interface TokenPayload {
    id: string;     // UUID como string
    email: string;
    cnpj: string;
    name: string;
    industry: string[];
    userType: string;
  }
  